x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/plan/logical/KeyedFilter.java

```diff
@@ -62,7 +62,9 @@ public Attribute tiebreaker() {
     public List<? extends NamedExpression> extractionAttributes() {
         List<NamedExpression> out = new ArrayList<>();
 
-        out.add(timestamp);
+        if (Expressions.isPresent(timestamp)) {
```

Comment:
Add a comment here for the scenario when the timestamp is not actually present, even though is a required request parameter.
Also, adjust the class level comment of this class which says "Used inside Join or Sequence".